src/backend/cpu/harris.cpp

```diff
@@ -166,7 +160,7 @@ unsigned harris(Array<float> &x_out, Array<float> &y_out, Array<float> &resp_out
     else {
         gaussian1D<convAccT>(h_filter, (int)filter_len, sigma);
     }
-    Array<convAccT> filter = createHostDataArray(filter_len, h_filter);
+    Array<convAccT> filter = createDeviceDataArray<convAccT>(dim4(filter_len), (const void*)h_filter);
```

Comment:
Can you make the changes in other files as well ? grep for `createHostDataArray` and make changes to avoid unnecessary copies. The two files that I could find are `where.cpp` and `orb.cpp` in cpu backend.